corehq/apps/saved_reports/tasks.py

```diff
@@ -86,7 +87,11 @@ def purge_old_scheduled_report_logs():
 @serial_task('queue_scheduled_reports', queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
 def queue_scheduled_reports():
     for report_id in create_records_for_scheduled_reports():
-        send_delayed_report(report_id)
+        try:
+            send_delayed_report(report_id)
+        except ResourceNotFound:
```

Comment:
`ResourceNotFound` is a generic exception that could be caused by any Couch query. I'm not very familiar with `send_delayed_report`, but is it possible that something else in there could cause that error to be thrown (and that should not be swallowed like this)? Would it make sense to catch `ResourceNotFound` on the one specific Couch lookup that gets the report and re-raise a new exception that will not possibly come from somewhere else in the code? Obviously that new exception would be caught and swallowed here.

Reply:
I'm still fairly conflicted over this. Ultimately, I think I found the responses on [when to use custom exceptions](https://stackoverflow.com/questions/57165380/when-should-i-declare-custom-exceptions) decent. Mostly, I want to avoid the proliferation of one-off errors and the appearance of errors-as-flow-control. What really bugs me about the code as written, however, is that `ResourceNotFound` exposes Couch into an area of code that should be indifferent to what database we use. Reluctantly, I'll wrap and re-throw a domain-specific exception.